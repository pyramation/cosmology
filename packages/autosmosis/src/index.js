export * from './assets';
export * from './clients';
export * from './driver/driver';