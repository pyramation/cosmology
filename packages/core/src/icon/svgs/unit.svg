<?xml version="1.0" encoding="UTF-8"?>
<svg width="1em" height="1em" viewBox="0 0 18 18" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
    <title>unit</title>
    <desc>Created with Sketch.</desc>
    <g id="unit" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M0.107142857,4 L17.8928571,4 C17.9520305,4 18,4.04663701 18,4.10416667 L18,13.8958333 C18,13.953363 17.9520305,14 17.8928571,14 L0.107142857,14 C0.0479694911,14 0,13.953363 0,13.8958333 L0,4.10416667 C0,4.04663701 0.0479694911,4 0.107142857,4 Z M2.125,7 C2.05596441,7 2,7.04796949 2,7.10714286 L2,12.8928571 C2,12.9520305 2.05596441,13 2.125,13 L3.875,13 C3.94403559,13 4,12.9520305 4,12.8928571 L4,7.10714286 C4,7.04796949 3.94403559,7 3.875,7 L2.125,7 Z M5.125,8 C5.05596441,8 5,8.04663701 5,8.10416667 L5,12.8958333 C5,12.953363 5.05596441,13 5.125,13 L6.875,13 C6.94403559,13 7,12.953363 7,12.8958333 L7,8.10416667 C7,8.04663701 6.94403559,8 6.875,8 L5.125,8 Z M8.125,7 C8.05596441,7 8,7.04796949 8,7.10714286 L8,12.8928571 C8,12.9520305 8.05596441,13 8.125,13 L9.875,13 C9.94403559,13 10,12.9520305 10,12.8928571 L10,7.10714286 C10,7.04796949 9.94403559,7 9.875,7 L8.125,7 Z M11.125,8 C11.0559644,8 11,8.04663701 11,8.10416667 L11,12.8958333 C11,12.953363 11.0559644,13 11.125,13 L12.875,13 C12.9440356,13 13,12.953363 13,12.8958333 L13,8.10416667 C13,8.04663701 12.9440356,8 12.875,8 L11.125,8 Z M14.125,7 C14.0559644,7 14,7.04796949 14,7.10714286 L14,12.8928571 C14,12.9520305 14.0559644,13 14.125,13 L15.875,13 C15.9440356,13 16,12.9520305 16,12.8928571 L16,7.10714286 C16,7.04796949 15.9440356,7 15.875,7 L14.125,7 Z" id="Shape" fill="currentColor" fill-rule="nonzero"></path>
    </g>
</svg>