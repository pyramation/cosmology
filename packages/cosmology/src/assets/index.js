import json from './assetlist.json';

export const assets = json.assets;
export * from './helpers';
// https://raw.githubusercontent.com/osmosis-labs/assetlists/main/osmosis-1/osmosis-1.assetlist.json
