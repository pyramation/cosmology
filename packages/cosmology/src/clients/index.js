export * from './osmosis';
export * from './cosmos';
export * from './validator';