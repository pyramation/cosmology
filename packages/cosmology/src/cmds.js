
import _create_mnemonic_wallet from './commands/create-mnemonic-wallet';
import _decrypt from './commands/decrypt';
import _delegate from './commands/delegate';
import _encrypt from './commands/encrypt';
import _get_balances from './commands/get-balances';
import _get_cosmos_balance from './commands/get-cosmos-balance';
import _get_latest_block from './commands/get-latest-block';
import _get_pool_apr from './commands/get-pool-apr';
import _get_price from './commands/get-price';
import _get_staking_rewards from './commands/get-staking-rewards';
import _get_transaction from './commands/get-transaction';
import _ibc_transfer from './commands/ibc-transfer';
import _ibc_wip from './commands/ibc-wip';
import _join_pool from './commands/join-pool';
import _join from './commands/join';
import _list_apis from './commands/list-apis';
import _list_pools from './commands/list-pools';
import _list_prices from './commands/list-prices';
import _load_recipe from './commands/load-recipe';
import _lock_tokens from './commands/lock-tokens';
import _lock from './commands/lock';
import _reinvest_copy from './commands/reinvest-copy';
import _reinvest from './commands/reinvest';
import _salt_decrypt from './commands/salt-decrypt';
import _salt_encrypt from './commands/salt-encrypt';
import _salt_generate from './commands/salt-generate';
import _send from './commands/send';
import _shamir_decrypt from './commands/shamir-decrypt';
import _shamir_encrypt from './commands/shamir-encrypt';
import _swap_exact_amount_in from './commands/swap-exact-amount-in';
import _swap from './commands/swap';
import _wallet_from_mnemonic from './commands/wallet-from-mnemonic';
const Commands = {};
Commands['create-mnemonic-wallet'] = _create_mnemonic_wallet;
Commands['decrypt'] = _decrypt;
Commands['delegate'] = _delegate;
Commands['encrypt'] = _encrypt;
Commands['get-balances'] = _get_balances;
Commands['get-cosmos-balance'] = _get_cosmos_balance;
Commands['get-latest-block'] = _get_latest_block;
Commands['get-pool-apr'] = _get_pool_apr;
Commands['get-price'] = _get_price;
Commands['get-staking-rewards'] = _get_staking_rewards;
Commands['get-transaction'] = _get_transaction;
Commands['ibc-transfer'] = _ibc_transfer;
Commands['ibc-wip'] = _ibc_wip;
Commands['join-pool'] = _join_pool;
Commands['join'] = _join;
Commands['list-apis'] = _list_apis;
Commands['list-pools'] = _list_pools;
Commands['list-prices'] = _list_prices;
Commands['load-recipe'] = _load_recipe;
Commands['lock-tokens'] = _lock_tokens;
Commands['lock'] = _lock;
Commands['reinvest-copy'] = _reinvest_copy;
Commands['reinvest'] = _reinvest;
Commands['salt-decrypt'] = _salt_decrypt;
Commands['salt-encrypt'] = _salt_encrypt;
Commands['salt-generate'] = _salt_generate;
Commands['send'] = _send;
Commands['shamir-decrypt'] = _shamir_decrypt;
Commands['shamir-encrypt'] = _shamir_encrypt;
Commands['swap-exact-amount-in'] = _swap_exact_amount_in;
Commands['swap'] = _swap;
Commands['wallet-from-mnemonic'] = _wallet_from_mnemonic;

  export { Commands }; 

  