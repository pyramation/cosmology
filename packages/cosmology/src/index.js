export * from './assets';
export * from './clients';
export * from './utils/osmo';
export * from './utils/chain';
export * from './messages';
