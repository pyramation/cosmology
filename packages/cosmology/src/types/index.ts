
export interface Pool {
    id: string;
    name: string;
    address: string;
    displayPoolAssets: object[];
    poolAssets: object[];
    pricePerShareEn18: string;
    totalShares: Coin;
    totalValue: string;
    totalWeight: string;
}

export interface TradeRoute {
    poolId: string;
    tokenOutDenom: CoinDenom;
    tokenOutSymbol: CoinSymbol;
    tokenInSymbol: CoinSymbol;
    liquidity: number;
}

export interface Swap {
    trade: Trade;
    routes: TradeRoute[];
};

export interface Pair {
    pool_address: string;
    pool_id: string;
    base_name: string;
    base_symbol: CoinSymbol;
    base_address: CoinDenom;
    quote_name: string;
    quote_symbol: CoinSymbol;
    quote_address: CoinDenom;
    price: number;
    base_volume_24h: number;
    quote_volume_24h: number;
    volume_24h: number;
    volume_7d: number;
    liquidity: number;
    liquidity_atom: number;
}
export interface LockedPoolDisplay {
    amount: string;
    denom: CoinDenom;
    value: string;
    allocation: string;
    poolId: string;
}
export interface Coin {
    amount: string;
    denom: CoinDenom;
}
export interface CoinWeight {
    weight: string;
    type: 'coin' | 'pool';
    name: string;
    value: string | null;
    symbol: CoinSymbol | null;
    poolId: string | null;
    denom: CoinDenom;
    allocation: string;
}
export interface CoinValue {
    amount: string;
    denom: CoinDenom;
    displayAmount: string;
    value: string;
    symbol: CoinSymbol;
}
export interface PoolAllocation {
    name: string;
    denom: CoinDenom;
    amount: string | null;
    displayAmount: string | null;
    value: string;
    coins: CoinValue[];
}
export interface DisplayCoin {
    amount: string;
    symbol: CoinSymbol;
}
export interface Trade {
    sell: CoinValue
    buy: CoinValue;
    beliefValue: string;
}
export type CoinDenom = string;

export type CoinSymbol =
    'ATOM' |
    'OSMO' |
    'ION' |
    'AKT' |
    'DVPN' |
    'IRIS' |
    'CRO' |
    'XPRT' |
    'REGEN' |
    'IOV' |
    'NGM' |
    'EEUR' |
    'JUNO' |
    'LIKE' |
    'UST' |
    'LUNA' |
    'BCNA' |
    'SCRT' |
    'MED';

export type CoinGeckoToken =
    'cosmos' |
    'osmosis' |
    'ion' |
    'akash-network' |
    'sentinel' |
    'iris-network' |
    'crypto-com-chain' |
    'persistence' |
    'regen' |
    'starname' |
    'e-money' |
    'e-money-eur' |
    'juno-network' |
    'likecoin' |
    'terrausd' |
    'terra-luna' |
    'bitcanna' |
    'terra-krw' |
    'secret' |
    'medibloc' |
    'comdex' |
    'cheqd-network' |
    'vidulum';

export interface PriceHash {
    [key: CoinDenom]: number;
}

export interface CoinGeckoUSD {
    usd: number;
};

export type CoinGeckoUSDResponse = Record<CoinGeckoToken, CoinGeckoUSD>


export interface ValidatorToken {
    price: number;
    denom: CoinDenom;
    symbol: CoinSymbol;
    liquidity: number;
    liquidity_24h_change: number;
    volume_24h: number;
    volume_24h_change: number;
    name: string;
};

export interface ValidatorTokenPrice {
    price: number;
}

export interface ValidatorCoinApr {
    start_date: string;
    denom: CoinDenom;
    symbol: CoinSymbol;
    apr_1d: number;
    apr_7d: number;
    apr_14d: number;
}

export interface ValidatorPoolApr {
    pool_id: number;
    apr_list: ValidatorCoinApr[]
}

export interface ValidatorPair {
    pool_address: string;
    pool_id: string;
    base_name: string;
    base_symbol: CoinSymbol;
    base_address: CoinDenom;
    quote_name: string;
    quote_symbol: CoinSymbol;
    quote_address: CoinDenom;
    price: number;
    base_volume_24h: number;
    quote_volume_24h: number;
    volume_24h: number;
    volume_7d: number;
    liquidity: number;
    liquidity_atom: number;
}

