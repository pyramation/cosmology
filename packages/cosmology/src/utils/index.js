export * from './chain';
export * from './helpers';
export * from './osmo';
export * from './print';
export * from './prompt';
export * from './prompts';
export * from './wallet';
export * from './gas';
