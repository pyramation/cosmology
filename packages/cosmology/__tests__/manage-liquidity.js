// @ts-nocheck
import pricesFixture from '../__fixtures__/coingecko/api/v3/simple/price/data.json';
import pairsFixture from '../__fixtures__/validator/pairs/v1/summary/data.json';
import poolsFixture from '../__fixtures__/lcd/osmosis/gamm/v1beta1/pools/data.json';
import cases from 'jest-in-case';
import {
  convertGeckoPricesToDenomPriceHash,
  symbolsAndDisplayValuesToCoinsArray,
  getTradesRequiredToGetBalances
} from '../src/utils/osmo';
import { CoinPretty, Dec, DecUtils, Int, IntPretty } from '@keplr-wallet/unit';

import { prettyPool } from '../src';
import {
  baseUnitsToDollarValue,
  dollarValueToDenomUnits,
  dollarValueToDisplayUnits,
  calculateShareOutAmount,
  displayUnitsToDollarValue,
  calculateMaxCoinsForPool
} from '../src/utils/chain';

const fakePools = [
  {
    '@type': '/osmosis.gamm.v1beta1.Pool',
    address: 'osmo1mw0ac6rwlp5r8wapwk3zs6g29h8fcscxqakdzw9emkne6c8wjp9q0t3v8t',
    id: '1010101',
    poolParams: {
      swapFee: '0.003000000000000000',
      exitFee: '0.000000000000000000',
      smoothWeightChangeParams: null
    },
    future_pool_governor: '24h',
    totalShares: {
      denom: 'gamm/pool/1010101',
      amount: '330228752041469100486167962'
    },
    poolAssets: [
      {
        token: {
          denom:
            'ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4',
          amount: '6196127370623'
        },
        weight: '357913941333333'
      },
      {
        token: {
          denom:
            'ibc/E6931F78057F7CC5DA0FD6CEF82FF39373A6E0452BF1FD76910B93292CF356C1',
          amount: '24164665575940'
        },
        weight: '357913941333333'
      },
      {
        token: {
          denom: 'uosmo',
          amount: '24164665575940'
        },
        weight: '357913941333333'
      }
    ],
    totalWeight: '1073741824000000'
  }
];

const prices = convertGeckoPricesToDenomPriceHash(pricesFixture);
cases(
  'joins',
  (opts) => {
    const balances = symbolsAndDisplayValuesToCoinsArray(opts.balances);
    const pool = poolsFixture.pools
      .concat(fakePools)
      .find((pool) => pool.id == opts.poolId);
    const poolInfo = prettyPool(pool, { includeDetails: true });
    const info = calculateMaxCoinsForPool(prices, poolInfo, balances);
    expect(info).toMatchSnapshot();
  },
  [
    {
      name: 'simple',
      balances: [
        {
          symbol: 'ATOM',
          amount: 0.29
        },
        {
          symbol: 'AKT',
          amount: 3
        }
      ],
      poolId: '4'
    },
    {
      name: 'complex',
      balances: [
        {
          symbol: 'ATOM',
          amount: 0.29
        },
        {
          symbol: 'AKT',
          amount: 3
        },
        {
          symbol: 'CRO',
          amount: 3
        },
        {
          symbol: 'OSMO',
          amount: 3
        }
      ],
      poolId: '1010101'
    }
  ]
);
