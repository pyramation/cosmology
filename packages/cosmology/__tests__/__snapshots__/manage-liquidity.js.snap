// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joins complex 1`] = `
Array [
  Object {
    "amount": "607771.8232044198",
    "denom": "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4",
  },
  Object {
    "amount": "300000000",
    "denom": "ibc/E6931F78057F7CC5DA0FD6CEF82FF39373A6E0452BF1FD76910B93292CF356C1",
  },
  Object {
    "amount": "120886.48351648351",
    "denom": "uosmo",
  },
]
`;

exports[`joins simple 1`] = `
Array [
  Object {
    "amount": "3000000",
    "denom": "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4",
  },
  Object {
    "amount": "76365.62182016727",
    "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
  },
]
`;
